### PR TITLE
Give more information dependencies were not resolved

### PR DESCRIPTION
**Pull request by fridex**

### CODE REVIEW

File: thoth/adviser/exceptions.py [hunk 1]
```diff
@@ -102,3 +108,19 @@ def __init__(self, *args: Any, unresolved: List[str]) -> None:
         """Capture unresolved dependencies in this exception."""
         super().__init__(*args)
         self.unresolved = unresolved
+
+    def to_dict(self) -> Optional[Dict[str, str]]:
+        """Convert unresolved dependencies exception to the user."""
+        return {
+            "ERROR": f"No dependencies found for {', '.join(f'{dep!r}' for dep in self.unresolved)}; "
```
Comment (goern) [inline thoth/adviser/exceptions.py]: can we say why we have not found any dependencies?

File: thoth/adviser/run.py [hunk 2]
```diff
@@ -73,13 +74,20 @@ def subprocess_run(
...
+            result_dict.update(dict(
+                error=True,
+                error_msg=str(exc),
+                report=dict(ERROR="An error occurred, see logs for more info")
+            ))
 
+        # Always submit results, even on error.
         print_func(time.monotonic() - start_time, result_dict)
-        sys.exit(0)
+        sys.exit(int(result_dict["error"]))
```
Comment (goern) [inline thoth/adviser/run.py]: this will yield 1 on error?

Comment (fridex) [reply thoth/adviser/run.py]: yes, `sys.exit(1)` on error so we will propagate failure (e.g. later schedule solvers based on an error in workflows)

File: thoth/adviser/exceptions.py (hunk 1 above)
Comment (fridex) [reply thoth/adviser/exceptions.py]: Yes, I think so :+1:

Comment (goern) [reply thoth/adviser/exceptions.py]: good, lets do it :)

Comment (fridex) [reply thoth/adviser/exceptions.py]: Done it in b30d64 :)

**Comment by fridex** [comment]
Rebased, review comments addressed. @bissenbay @goern PTAL, I would like to propagate this to stage.

**Pull request merged**